lib/environment.rb

```diff
@@ -321,24 +321,27 @@ def get_routing_details(user:, project:)
       route = service.expose(user: user)
 
       fqdn = route.dns(by: user)
-      opts[:router_subdomain] = fqdn.split('.',2)[1]
-      opts[:router_ips] = Common::Net.dns_lookup(fqdn, multi: true)
-
+      if obj == 'subdomain'
+        opts[:router_subdomain] = fqdn.split('.',2)[1]
+      end
+      if obj == 'ips'
+        opts[:router_ips] = Common::Net.dns_lookup(fqdn, multi: true)
+      end
```

Comment:
Perhaps use if/else so we do not need to do if check twice every time.

Reply:
@lihongan @liangxia's suggestion should be easy to implement.  Please correct amend this PR instead of filing another one.

Reply:
@liangxia @pruan-rht Thank you all, I've updated and use if/else.

Reply:
@lihongan do you foresee `get_routing_details` be called with object type other than `subdomain` and `ips`?

Reply:
@pruan-rht  No, I don't see any other caller. The two object type is added by this PR. 
The problem is the old `get_routing_details` gets both `router_ips` and `router_subdomain` at the same time even I just need one of them.
Another possible fix is removing `get_routing_details` and implementing each function in `router_ips(user:, project:)` and `router_default_subdomain(user:, project:)`, the two callers I can find.

Reply:
@lihongan Ok, I would suggest we separate them into two wrapper functions both of which call the base function but with the parameter pass in.  IMO, this will make it clearer to the user just by reading the method name instead of having to look up what parameters are accepted.  WDYT?

Reply:
No problem. Both solutions are accepted.
My requirement is very simple: just return what I want when I getting `router_ips` or `router_subdomain`. Don't get both even I just need one of them.
In our scenarios, most cases are just need one of them.